src/GafferSceneUI/Inspector.cpp

```diff
@@ -301,6 +310,303 @@ void Inspector::editScopeInputChanged( const Gaffer::Plug *plug )
+			boost::bind( &HistoryPath::inspectorDirtied, this, ::_1 )
+		);
+		m_pathChangedConnection = pathChangedSignal().connect(
+			boost::bind( &HistoryPath::pathChanged, this, ::_1 )
+		);
+	}
+
+	emitPathChanged();
+}
+
+Inspector::HistoryPath::~HistoryPath()
+{
+
+}
+
+void Inspector::HistoryPath::propertyNames( std::vector<InternedString> &names, const Canceller *canceller) const
+{
+	Path::propertyNames( names );
+
+	if( isLeaf() )
+	{
+		names.push_back( g_valuePropertyName );
+		names.push_back( g_operationPropertyName);
+		names.push_back( g_acquiredEditPropertyName );
+		names.push_back( g_editWarningPropertyName );
+	}
+}
+
+ConstRunTimeTypedPtr Inspector::HistoryPath::property( const InternedString &name, const Canceller *canceller) const
+{
+
+	if( isLeaf() && name == g_namePropertyName )
+	{
+		// Remove the plug name from the end.
+		using Tokenizer = boost::tokenizer<boost::char_separator<char>>;
+
+		std::string nodeString;
+
+		ConstStringDataPtr nameData = runTimeCast<const StringData>( Path::property( g_namePropertyName, canceller ) );
+		std::string name = nameData->readable();
+
+		Tokenizer tokens( name, boost::char_separator<char>( "." ) );
+
+		for( Tokenizer::const_iterator it = tokens.begin(), eIt = tokens.end(); it != eIt; ++it )
+		{
+			Tokenizer::const_iterator next = it;
+			next++;
+			if( next == tokens.end() )
+			{
+				continue;
+			}
+			if( nodeString.size() )
+			{
+				nodeString += ".";
+			}
+			nodeString += *it;
+		}
+
+		return new StringData( nodeString );
+	}
+
+	if(
+		isLeaf() && (
+			name == g_valuePropertyName ||
+			name == g_operationPropertyName ||
+			name == g_acquiredEditPropertyName ||
+			name == g_editWarningPropertyName
+		)
+	)
+	{
+		const std::string pathName = names()[0].string();
+
+		Context::Scope scope( m_context.get() );
+
+		SceneAlgo::History::ConstPtr history = m_inspector->history();
+		const SceneAlgo::History *currentHistory = history.get();
+
+		ScenePlug *targetPlug = m_scriptNode.get()->descendant<ScenePlug>( pathName );
+
+		assert( targetPlug );
+
+		while( true )
+		{
+			std::string editWarning;
+
+			Context::Scope currentScope( currentHistory->context.get() );
+
+			if( ValuePlugPtr immediateSource = m_inspector->source( currentHistory, editWarning ) )
+			{
+				if( currentHistory->scene == targetPlug )
+				{
+					ValuePlug *source = static_cast<ValuePlug *>( spreadsheetAwareSource( immediateSource.get() ) );
+					const std::string nodeName = currentHistory->scene->relativeName( m_scriptNode.get() );
+
+					if( name == g_valuePropertyName )
+					{
+						return runTimeCast<const IECore::Data>( m_inspector->value( currentHistory ) );
+					}
+					else if( name == g_operationPropertyName )
+					{
+						if( auto tweakPlug = runTimeCast<const TweakPlug>( source ) )
+						{
+							return new IntData( tweakPlug->modePlug()->getValue() );
+						}
+						return new IntData( TweakPlug::Mode::Create );
+					}
+					else if( name == g_acquiredEditPropertyName )
```

Comment:
Seems like `g_sourcePropertyName` would make more sense now?

Reply:
Fixed in e6bd27f7be95b6d09d20f43b212b2bbdac71302b